homeassistant/components/octoprint/__init__.py

```diff
@@ -171,9 +180,74 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     hass.config_entries.async_setup_platforms(entry, PLATFORMS)
 
+    async def async_pause_print(call: ServiceCall) -> None:
+        """Pauses the active print."""
+        client = async_get_client_for_service_call(hass, call)
+        printer = await client.get_printer_info()
+        if printer.state.flags.printing:
+            await client.pause_job()
+        else:
+            raise InvalidPrinterState("Printer is not printing")
```

Comment:
I haven't been able to find any official guidance on this in a brief search of the docs, but, in my opinion, a service call should only fail if it either can't connect to the printer or if the printer can't be put in the requested state. If the printer is *already* in the requested state (in this case, not printing), that shouldn't be a failure. Otherwise, automations that use the service become more complicated for not much benefit.

Same goes for the other two. Stop should succeed if the printer is already stopped. Resume should succeed if it's already printing (though I can see an argument for failing if it is stopped with nothing to resume).

Reply:
Makes sense to me, updated the PR to not raise an exception if the machine is in the requested state(IE stopped, paused)